src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.IO;
+using System.Text;
+using Microsoft.AspNet.Mvc.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Wraps the supplied <see cref="Stream"/> to prevent it from flushing (which can cause response to be sent
+    /// in Chunked encoding) and disposing (as the hosting layers own the stream).
+    /// Also wraps the supplied <see cref="Encoding"/> to prevent writing the preamble or BOM bytes to the response.
+    /// </summary>
+    public class HttpResponseStreamWriter : StreamWriter
```

Comment:
Not sure it matters. I think today we're actually double-buffering everything because of this. Talk to @Tratcher